graphium/nn/architectures/global_architectures.py

```diff
@@ -1278,6 +1301,95 @@ def set_max_num_nodes_edges_per_graph(self, max_nodes: Optional[int], max_edges:
 
         self.task_heads.set_max_num_nodes_edges_per_graph(max_nodes, max_edges)
 
+    def modify_architecture(self, cfg):
+
+        cfg_finetune = cfg['trainer']['finetuning']
+        choice = cfg_finetune['choice']
+        task = cfg_finetune['task']
+
+        if cfg_finetune['module'] == 'task_heads':
+            self.task_heads.task_heads = nn.ModuleDict({task: self.task_heads.task_heads[choice]})
+
+            self.task_heads.task_heads[task].drop_layers(cfg_finetune['depth'])
+
+            in_dim = self.task_heads.task_heads[task].layers[-1].out_dim
+            added_layers = FeedForwardNN(in_dim, **cfg_finetune['added_layers']).layers
+
+            self.task_heads.task_heads[task].add_layers(added_layers)
+
+            self.task_heads.task_heads_kwargs = {task: self.task_heads.task_heads_kwargs[choice]}
+            self.task_heads.task_heads_kwargs[task].update(
+                {
+                    'out_dim': cfg_finetune['added_layers']['out_dim'],
+                    'depth': self.task_heads.task_heads_kwargs[task]['depth'] + cfg_finetune['added_layers']['depth'],
+                    'name': 'NN-' + task
+                }
+            )
+
+        else:
+            raise NotImplementedError()
+
+    def overwrite_with_pretrained(self, cfg, pretrained_model):
+
+        cfg_finetune = cfg['finetuning']
+        task_head_from_pretrained = cfg_finetune['task_head_from_pretrained']
+        task = cfg_finetune['task']
+        added_depth = cfg_finetune['added_depth']
+
+        for module in ['pre_nn', 'pre_nn_edges', 'gnn', 'graph_output_nn', 'task_heads']:
+
+            if module == cfg_finetune['module_from_pretrained']:
+                break
```

Comment:
`break` (exit the loop entirely) or `continue` (skip a single loop iteration)?

Reply:
Yes, because we only want to fully overwrite modules BEFORE the module we finetune from. The finetuning module is overwritten partially and modules afterwards are not overwritten (they keep their random initialization).

Reply:
Ah check! I understand now, thanks!